tests/test_admin.py

```diff
@@ -482,3 +482,49 @@ def test_versions_filters_all_state(self):
             transform=lambda x: x.pk,
             ordered=False,
         )
+
+
+class ContentExpiryCsvExportTestCase(CMSTestCase):
+    def test_export_button_endpoint_response_is_a_csv(self):
+        """
+        Valid csv file is returned from the admin export endpoint
+        """
+        date = datetime.datetime.now() - datetime.timedelta(days=5)
+        PollContentExpiryFactory(expires=date, version__state=DRAFT)
+
+        version_selection = "?state=_all_"
+        export_endpoint = "export_csv/"
+        admin_endpoint = self.get_admin_url(ContentExpiry, "changelist")
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(admin_endpoint + export_endpoint + version_selection)
+
+        # Endpoint is returning 200 status code
+        self.assertEqual(response.status_code, 200)
+        # Response contains a csv file
+        self.assertEquals(
+            response.get('Content-Disposition'),
+            "attachment; filename={}.csv".format("djangocms_content_expiry.contentexpiry")
+        )
+
+    def test_filter_boundaries_respected_in_export(self):
+        """
+        Export respects applied filters.
+        Only content in view should be exported.
+        """
+        date = datetime.datetime.now() - datetime.timedelta(days=5)
+        # Create content expiry records for draft and published
+        PollContentExpiryFactory(expires=date, version__state=DRAFT)
+        PollContentExpiryFactory(expires=date, version__state=PUBLISHED)
+
+        # When draft is selected only the draft entries should be shown
+        version_selection = f"?state={DRAFT}"
+        export_endpoint = "export_csv/"
+        admin_endpoint = self.get_admin_url(ContentExpiry, "changelist")
+
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(admin_endpoint + export_endpoint + version_selection)
+
+        response_content = response.content.decode()
+        # Published content should not be available in exported data if the filter is set to display draft only
+        self.assertNotIn('Published', response_content)
```

Comment:
We should ideally check each filter that we can control. For maintainability we would split them somehow, probably a test for each filter with most of the reusable bits in the TestCase setup().

Think of it as you're a dev in the future adding a new filter, you should be able to add your new feature and tests without affecting previous tests. It should also be simple to maintain too, you're the dev making a change to the filters, the test changes shouldn't impact the other filters. 

Example:

```
class ContentExpiryCsvExportFilterSettingsTestCase(CMSTestCase):

    def setUp(self):
        ... repetitive and reusable bits required for every test

    def test_version_filter_boundaries_in_export():
        ...

    def test_author_filter_boundaries_in_export():
        ...

    def test_content_type_filter_boundaries_in_export():
        ...

     def test_mynewfuture_filter_boundaries_in_export():
        ...
```